### PR TITLE
Alternative `minimum_image` code

### PR DESCRIPTION
**Pull request by stanmoore1**
**Summary**

The code for `minimum_image` seems a little complex with the while loop, and is only good for 16 box lengths. Here is alternative code for consideration, which is good for up to 2 billion box lengths in a single step 😄 Could go even further with a 64 bit integer but seems unnecessary.

**Related Issue(s)**

None

**Author(s)**

Stan Moore (SNL)

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

Yes

### CODE REVIEW

**Comment by stanmoore1** [comment]
@akohlmey @sjplimp I'm not saying we should necessarily merge this, but wanted to get your thoughts on this.

**Comment by akohlmey** [comment]
@stanmoore1 you have to know the history of this function to understand why it is the way it has become.

The code was initially different (I think it used a modulus) and loops were used to make it faster and more reliable by avoiding divisions. For a well behaved simulation, there are only 1 or 2 trips of the loop.

But then we added the limit because there was an increasing number of reports of LAMMPS "getting stuck" without any indication of where and why. Those would have crazy coordinates (like > 1.0e100 or NaN) where the loop would not work due to the finite precision of floating point math. So the limit is a good thing because it reports bogus simulations sooner.

Bottom line: is using the loops in the current format (still) faster than the alternatives with a division or modulus?

**Comment by stanmoore1** [comment]
The source of all absolute truth suggests this is the fastest way for x86-64: https://en.wikipedia.org/wiki/Periodic_boundary_conditions. But I haven't measured yet.

**Comment by stanmoore1** [comment]
Though in the Wikipedia description they are storing `1/xprd`, etc. and only update it when the box length changes to avoid division.

**Comment by stanmoore1** [comment]
Closing for now.
